src/sentry/replays/consumers/buffered/factory.py

```diff
@@ -0,0 +1,32 @@
+"""Session Replay recording consumer strategy factory.
+
+This module exists solely to abstract the bootstraping process of the application and runtime in
+`sentry/consumers/__init__.py`.
+"""
+
+from collections.abc import Mapping
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.types import Commit as ArroyoCommit
+from arroyo.types import Partition
+
+from sentry.replays.consumers.buffered.consumer import Flags, recording_runtime
+from sentry.replays.consumers.buffered.platform import PlatformStrategy
+
+
+class PlatformStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+
+    def __init__(self, max_buffer_length: int, max_buffer_wait: int, max_workers: int) -> None:
+        self.flags: Flags = {
+            "max_buffer_length": max_buffer_length,
+            "max_buffer_wait": max_buffer_wait,
+            "max_workers": max_workers,
+        }
+
+    def create_with_partitions(
+        self,
+        commit: ArroyoCommit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        return PlatformStrategy(commit=commit, flags=self.flags, runtime=recording_runtime)
```

Comment:
Wouldn't it be considerably simpler to model this consumer as a sequence of these Arroyo operators:
- Batch the messages with the Reduce primitive (https://getsentry.github.io/arroyo/strategies/reduce.html#reduce-fold)
- Run Task in either multi threading or multi processing (https://getsentry.github.io/arroyo/strategies/reduce.html#reduce-fold) to perform the parsing and the `commit_recording_message` ?

Modeling the system this way would:
- allow the parallelism either via processes and threads without application logic changes
- guarantee a pipeline approach that allows the batching step to keep batching new messages while the worker thread performs its work.
- hide offset management entirely from the application code.